### PR TITLE
[MRG+1] Pull Request template

### PR DESCRIPTION
**Pull request by story645**
I thought that parts of the discussion in #8456 were well worth exposing to new contributors and @phobson suggested that a PR template might be a good place to do it and worthwhile as a way to maybe improve initial PRs. 

CC: @tacaswell, @WeatherGod, @phobson

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+- [ ] New features are documented-and have an example if plot related
+- [ ] Documentation is [sphinx](http://matplotlib.org/devdocs/devel/documenting_mpl.html) and [numpydoc](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt) compliant
+- [ ] Related issues and pull requests [are listed](https://help.github.com/articles/autolinked-references-and-urls/)
+
+** Major Changes **
+- [ ] major new feature: added an entry to [doc/users/whats_new.rst](../doc/users/whats_new).
+- [ ] change the API in a backward-incompatible way: documented it in [doc/api/api_changes.rst](../doc/api/api_changes)
+
+We understand that PRs can sometimes feel like a guantlet, especially as the reviews start coming in. Please let us know 
+if the reviews are unclear or the recommended next step is too ardous, and if you'd like help in addressing a reveiwers
```
Comment (WeatherGod) [inline .github/PULL_REQUEST_TEMPLATE.md]: arduous

Comment (WeatherGod) [inline .github/PULL_REQUEST_TEMPLATE.md]: reviewer's

**Comment by WeatherGod** [comment]
Could we also have a comment to leave a descriptive title? I *hate* titles that say "address such-and-such PR".

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+**[PR Checklist](http://matplotlib.org/devel/coding_guide.html#pull-request-checklist)**
+- [ ] Code is [tested](http://matplotlib.org/devel/testing.html#testing) using[Py.test](https://docs.pytest.org/en/latest/) style tests) on Py2 and Py3
+- [ ] Code is [PEP 8](https://www.python.org/dev/peps/pep-0008/) compliant 
+- [ ] New features are documented-and have an example if plot related
+- [ ] Documentation is [sphinx](http://matplotlib.org/devdocs/devel/documenting_mpl.html) and [numpydoc](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt) compliant
+- [ ] Related issues and pull requests [are listed](https://help.github.com/articles/autolinked-references-and-urls/)
+
+** Major Changes **
+- [ ] major new feature: added an entry to [doc/users/whats_new.rst](../doc/users/whats_new).
+- [ ] change the API in a backward-incompatible way: documented it in [doc/api/api_changes.rst](../doc/api/api_changes)
```
Comment (WeatherGod) [inline .github/PULL_REQUEST_TEMPLATE.md]: documented --> document

Comment (WeatherGod) [inline .github/PULL_REQUEST_TEMPLATE.md]: oh, nevermind, I see it is all supposed to be past tense.

**Comment by QuLogic** [comment]
Something to keep in mind is that this will appear in unrendered form in the description, thus it should be made as brief as possible (oftentimes reporters don't even read the issue template).

Relatedly, we shouldn't add too many links because they aren't rendered and you need to copy and paste to view what's in them instead of a simple click, so people won't look at them.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+Thank you so much for your PR! To help us review, fill out the form to the best of your ability. Also, please make use of the development guide at http://matplotlib.org/devdocs/devel/index.html
+
+<!--- Provide a general summary of your changes in the Title above, for example "Raises ValueError on Non-Numeric Input to set_xlim". Please avoid non-descriptive titles such as "Addresses issue #8576"-->
+
+## PR Summary
+<!--- Please provide at least 1-2 sentences describing the pull request in detail. Why is this change required? What problem does it solve? -->
+<!--- If it fixes an open issue, please link to the issue here. -->
+
+## PR Checklist
+- [ ] Code is tested using Py.test on Py2 and Py3
```
Comment (anntzer) [inline .github/PULL_REQUEST_TEMPLATE.md]: it's pytest (no dot) these days

it's may be a bit harsh to request testing on both py2 and py3 (plus you also need to check somewhat antique versions of numpy) so I'd just say "unit tests are present" (travis will catch failures anyways)  (for example I typically only run tests on py3.6+latest numpy and just let travis tell me if anything else needs to be corrected)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+- [ ] Code is PEP 8 compliant 
+- [ ] New features are documented-and have an example if plot related
```
Comment (anntzer) [inline .github/PULL_REQUEST_TEMPLATE.md]: extra dash

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+- [ ] Documentation is sphinx and numpydoc compliant
+- [ ] major new feature: added an entry to doc/users/whats_new.rst
```
Comment (anntzer) [inline .github/PULL_REQUEST_TEMPLATE.md]: "For major new features", and below "For backwards incompatible changes to the API"

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+- [ ] change the API in a backward-incompatible way: documented it in doc/api/api_changes.rst
+
+We understand that PRs can sometimes feel like a guantlet, especially as the reviews start coming in. Please let us know
```
Comment (anntzer) [inline .github/PULL_REQUEST_TEMPLATE.md]: gauntlet

the line wrapping is a bit weird.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+## PR Summary
+<!--- Please provide at least 1-2 sentences describing the pull request in detail. Why is this change required? What problem does it solve? -->
+<!--- If it fixes an open issue, please link to the issue here. -->
+
+## PR Checklist
+- [ ] Has Pytest style unit tests
+- [ ] Code is PEP 8 compliant 
+- [ ] New features are documented, with examples if plot related
+- [ ] Documentation is sphinx and numpydoc compliant
+- [ ] Added an entry to doc/users/whats_new.rst if major new feature:
```
Comment (anntzer) [inline .github/PULL_REQUEST_TEMPLATE.md]: remove end colon.
suggest squashing after that (not much point in having 10 commits for this single file)

**Pull request merged**